public/script.js

```diff
@@ -1458,21 +1458,8 @@ async function Generate(type, automatic_trigger, force_name2) {
             storyString += appendToStoryString(charPersonality, power_user.disable_personality_formatting ? '' : 'Personality: ');
             storyString += appendToStoryString(Scenario, power_user.disable_scenario_formatting ? '' : 'Scenario: ');
         } else {
-            if (charDescription !== undefined) {
-                if (charPersonality.length > 0 && !power_user.disable_personality_formatting) {
-                    charPersonality = name2 + "'s personality: " + charPersonality;
-                }
-            }
-
             storyString += appendToStoryString(charDescription, '');
-
-            if (storyString.endsWith('\n')) {
-                storyString = storyString.slice(0, -1);
```

Comment:
Allowing `storyString` to conditionally not end with a newline is quite error-prone, because it requires additional checks further down the line. And indeed this has lead to a bug: with *Disable chat start formatting* and enough messages in the chat for `count_view_mes < topAnchorDepth` to evaluate to false, description and character name become conjoined. Before and after this PR:
```
DESCRIPTIONTest: Hello
[Test Elaborate speaker]
You: Lorem ipsum
```
```
DESCRIPTION
Test: Hello
[Test Elaborate speaker]
You: Lorem ipsum
```
This bug was also capable of doing this to After Chara part of world info and example messages.

Does the fix have any side-effects? Looking at how the final prompt is constructed, I don't think appending the newline unconditionally breaks anything. The worst thing that can happen is that there will be a single blank line after the description in a long chat. In a short chat (`count_view_mes < topAnchorDepth`) it would be there even without this PR.
```
finalPromt = worldInfoBefore + storyString + worldInfoAfter + afterScenarioAnchor + mesExmString + mesSendString + generatedPromtCache + promptBias
```